src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copyState = { ...state };
+  const previousStateVers = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        copyState = Object.assign(copyState, action.extraData);
+        previousStateVers.push({ ...copyState });
+        break;
+      }
+
+      case 'removeProperties': {
+        for (const keyToRemove of action.keysToRemove) {
+          delete copyState[keyToRemove];
+        }
+        previousStateVers.push({ ...copyState });
```

Comment:
Don't repeat yourself: you've repeated the logic for adding new copy 3 times, let's move it outside of the `switch` block and write only once